priv/templates/phoenix.gen.model/model.ex

```diff
@@ -3,7 +3,7 @@ defmodule <%= module %> do
 
   schema <%= inspect plural %> do
 <%= for {k, _} <- attrs do %>    field <%= inspect k %>, <%= inspect types[k] %><%= schema_defaults[k] %>
-<% end %><%= for {k, _, m, _} <- assocs do %>    belongs_to <%= inspect k %>, <%= m %>
+<% end %><%= for {k, k_id, m, _} <- assocs do %>    belongs_to <%= inspect k %>, <%= m %><%= if !String.ends_with?(Atom.to_string(k_id), "_id") do %>, define_field: false, foreign_key: <%= inspect k %><% end %>
```

Comment:
Hmm, perhaps not. Reading the comments (here: https://github.com/elixir-lang/ecto/blob/master/lib/ecto/schema.ex#L674), I was under the impression that, when false, define_field simply didn't set the foreign key automatically, but it could then be set properly with the foreign_key option. 

I did a quick new-project test and ran the automated tests, so thought it was working correctly. I'll do another test and make sure I didn't mess something up.